File: lib/models/responsable.dart
// lib/models/responsable.dart
import 'mascota.dart';

class Responsable {
  final int id;
  final String nombre;
  final String telefono;
  final String finca;
  final String zona; // Nuevo campo: vereda, centro poblado, barrio
  final String nombreZona; // Nuevo campo: nombre de la zona
  final String loteVacuna; // Nuevo campo: lote de vacuna
  final DateTime creado;
  final List<Mascota> mascotas;

  Responsable({
    required this.id,
    required this.nombre,
    required this.telefono,
    required this.finca,
    required this.zona,
    required this.nombreZona,
    required this.loteVacuna,
    required this.creado,
    required this.mascotas,
  });

  factory Responsable.fromJson(Map<String, dynamic> json) => Responsable(
    id: json['id'] as int? ?? 0,
    nombre: json['nombre'] as String? ?? '',
    telefono: json['telefono'] as String? ?? '',
    finca: json['finca'] as String? ?? '',
    zona: json['zona'] as String? ?? 'vereda',
    nombreZona: json['nombre_zona'] as String? ?? '',
    loteVacuna: json['lote_vacuna'] as String? ?? '',
    creado: DateTime.tryParse(json['creado'] as String? ?? '') ?? DateTime.now(),
    mascotas: (json['mascotas'] as List<dynamic>?)
        ?.map((e) => Mascota.fromJson(e as Map<String, dynamic>))
        .toList() ?? <Mascota>[],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'nombre': nombre,
    'telefono': telefono,
    'finca': finca,
    'zona': zona,
    'nombre_zona': nombreZona,
    'lote_vacuna': loteVacuna,
    'creado': creado.toIso8601String(),
    'mascotas': mascotas.map((m) => m.toJson()).toList(),
  };

  // Opciones disponibles para el campo zona
  static const List<String> opcionesZona = [
    'vereda',
    'centro poblado', 
    'barrio',
  ];
} 

File: src/models/Responsable.ts
import { Mascota, MascotaModel } from './Mascota';

export interface Responsable {
  id: number;
  nombre: string;
  telefono: string;
  finca: string;
  zona: 'vereda' | 'centro poblado' | 'barrio';
  nombreZona: string;
  loteVacuna: string;
  creado: Date;
  mascotas: Mascota[];
}

export class ResponsableModel {
  static readonly opcionesZona = ['vereda', 'centro poblado', 'barrio'] as const;

  static fromJson(json: any): Responsable {
    return {
      id: json.id || 0,
      nombre: json.nombre || '',
      telefono: json.telefono || '',
      finca: json.finca || '',
      zona: json.zona || 'vereda',
      nombreZona: json.nombre_zona || '',
      loteVacuna: json.lote_vacuna || '',
      creado: json.creado ? new Date(json.creado) : new Date(),
      mascotas: (json.mascotas || []).map((m: any) => MascotaModel.fromJson(m)),
    };
  }

  static toJson(responsable: Responsable): any {
    return {
      id: responsable.id,
      nombre: responsable.nombre,
      telefono: responsable.telefono,
      finca: responsable.finca,
      zona: responsable.zona,
      nombre_zona: responsable.nombreZona,
      lote_vacuna: responsable.loteVacuna,
      creado: responsable.creado.toISOString(),
      mascotas: responsable.mascotas.map(m => MascotaModel.toJson(m)),
    };
  }
} 